test/unix/test_lwt_bytes.ml

```diff
@@ -6,6 +6,33 @@ open Test
 
 let bytes_equal (b1:Bytes.t) (b2:Bytes.t) = b1 = b2
 
+let tcp_server_client_exchange server_logic client_logic =
+  let server_is_ready, notify_server_is_ready = Lwt.wait () in
+  let server () =
+    let sock = Lwt_unix.socket Lwt_unix.PF_INET Lwt_unix.SOCK_STREAM 0 in
+    let sockaddr = Lwt_unix.ADDR_INET (Unix.inet_addr_loopback, 0) in
+    Lwt_unix.bind sock sockaddr
+    >>= fun () ->
+    let server_address = Lwt_unix.getsockname sock in
+    let () = Lwt_unix.listen sock 5 in
+    Lwt.wakeup_later notify_server_is_ready server_address;
+    Lwt_unix.accept sock
+    >>= fun (fd_client, _) ->
+    server_logic fd_client
+    >>= fun _n -> Lwt_unix.close fd_client
+    >>= fun () -> Lwt_unix.close sock
+  in
+  let client () =
+    server_is_ready
+    >>= fun sockaddr ->
+    let sock = Lwt_unix.socket Lwt_unix.PF_INET Lwt_unix.SOCK_STREAM 0 in
+    Lwt_unix.connect sock sockaddr
+    >>= fun () ->
+    client_logic sock
+    >>= fun _n -> Lwt_unix.close sock
+  in
+  Lwt_main.run (Lwt.join [client (); server ()])
```

Comment:
Sorry for not noticing this earlier, but the testing framework already calls `Lwt_main.run`: https://github.com/ocsigen/lwt/blob/c1e48e4807c3742bad632dc5905c71a815e5f042/test/test.ml#L204. Nested `Lwt_main.run` is not allowed, so this should just do the `join` without wrapping it in a `Lwt_main.run` call, and the code that calls this function should do `tcp_server_client_exchange >>= fun () -> ...` to wait for this to complete.